price-feeder/oracle/provider/mexc.go

```diff
@@ -0,0 +1,504 @@
+}
+
+// subscribeTickers subscribe to the ticker channel for all currency pairs.
+func (p *MexcProvider) subscribeTickers(cps ...types.CurrencyPair) error {
+	pairs := make([]string, len(cps))
+
+	for i, cp := range cps {
+		pairs[i] = currencyPairToMexcPair(cp)
+	}
+
+	return p.subscribePairs(pairs...)
+}
+
+// subscribeCandles subscribe to the candle channel for all currency pairs.
+func (p *MexcProvider) subscribeCandles(cps ...types.CurrencyPair) error {
+	pairs := make([]string, len(cps))
+
+	for i, cp := range cps {
+		pairs[i] = currencyPairToMexcPair(cp)
+	}
+
+	return p.subscribePairs(pairs...)
+}
+
+// subscribedPairsToSlice returns the map of subscribed pairs as a slice.
+func (p *MexcProvider) subscribedPairsToSlice() []types.CurrencyPair {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	return types.MapPairsToSlice(p.subscribedPairs)
+}
+
+func (p *MexcProvider) getTickerPrice(key string) (types.TickerPrice, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	ticker, ok := p.tickers[key]
+	if !ok {
+		return types.TickerPrice{}, fmt.Errorf("mexc provider failed to get ticker price for %s", key)
+	}
+
+	return ticker, nil
+}
+
+func (p *MexcProvider) getCandlePrices(key string) ([]types.CandlePrice, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	candles, ok := p.candles[key]
+	if !ok {
+		return []types.CandlePrice{}, fmt.Errorf("failed to get candle prices for %s", key)
+	}
+
+	return candles, nil
+}
+
+func (p *MexcProvider) messageReceived(messageType int, bz []byte) {
+	if messageType != websocket.TextMessage {
+		return
+	}
+
+	var (
+		tickerResp MexcTickerResponse
+		tickerErr  error
+		candleResp MexcCandleResponse
+		candleErr  error
+	)
+
+	tickerErr = json.Unmarshal(bz, &tickerResp)
+	for _, cp := range p.subscribedPairs {
+		mexcPair := currencyPairToMexcPair(cp)
+		if tickerResp.Symbol[mexcPair].LastPrice != 0 {
+			p.setTickerPair(
+				mexcPair,
+				tickerResp.Symbol[mexcPair],
+			)
+			telemetry.IncrCounter(
+				1,
+				"websocket",
+				"message",
+				"type",
+				"ticker",
+				"provider",
+				string(ProviderMexc),
+			)
+			return
+		}
+	}
+
+	candleErr = json.Unmarshal(bz, &candleResp)
+	if candleResp.Metadata.Close != 0 {
+		p.setCandlePair(candleResp)
+		telemetry.IncrCounter(
+			1,
+			"websocket",
+			"message",
+			"type",
+			"candle",
+			"provider",
+			string(ProviderMexc),
+		)
+		return
+	}
+
+	if tickerErr != nil || candleErr != nil {
+		p.logger.Error().
+			Int("length", len(bz)).
+			AnErr("ticker", tickerErr).
+			AnErr("candle", candleErr).
+			Msg("mexc: Error on receive message")
+	}
+}
+
+func (p *MexcProvider) setTickerPair(symbol string, ticker MexcTickerData) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	price, err := sdk.NewDecFromStr(strconv.FormatFloat(ticker.LastPrice, 'f', 5, 64))
+	if err != nil {
+		p.logger.Warn().Err(err).Msg("mexc: failed to parse ticker price")
+	}
+	volume, err := sdk.NewDecFromStr(strconv.FormatFloat(ticker.Volume, 'f', 5, 64))
```

Comment:
let's create a helper function for that:

```
func DecFromFloat(f float64) sdk.Dec
```

let's put it in `util/coin/dec.go`. I've created that package in https://github.com/umee-network/umee/pull/1326 but you can safely created that file without creating any conflict.